### PR TITLE
Replaced the Deprecated .substr with .substring.

### PR DESCRIPTION
**Pull request by Harry-Hopkinson**
This PR replaces the deprecated .substr with .substring.
All local tests passing with these changes.
I have accessed and checked all of the changes for each change.

### CODE REVIEW

**Review by Harry-Hopkinson — COMMENTED** [review]
I have assessed all changes to each files and all follow the replace .substr with .substring procedure.

File: build/gulpfile.vscode.web.js [hunk 1]
```diff
@@ -94,7 +94,7 @@ const createVSCodeWebProductConfigurationPatcher = (product) => {
 				commit,
 				date: buildDate
 			});
-			return content.replace('/*BUILD->INSERT_PRODUCT_CONFIGURATION*/', productConfiguration.substr(1, productConfiguration.length - 2) /* without { and }*/);
```
Comment (JacksonKearl) [inline build/gulpfile.vscode.web.js:97]: These are different, `"{ab}".substr(1,"{ab}".length - 2) === "ab"` , `"{ab}".substring(1,"{ab}".length - 2) === "a"`.

Comment (Harry-Hopkinson) [reply build/gulpfile.vscode.web.js:97]: Shall I revert to .substr instead?

**Comment by JacksonKearl** [comment]
This PR is far too large to be effectively reviewed, merging it would cause lot's of conflicts in addition to the behaviour changes I pointed out above. Please see our [contributing guidelines' section on PR's](https://github.com/microsoft/vscode/wiki/How-to-Contribute#pull-requests) for more info. 

While `substr` is technically deprecated, it's not going anywhere, so there's no pressing need to convert our usages over. I have heard reports that `substring` might be faster than `substr` in some cases. If you can find evidence this is true, consider opening an issue with more details and component owners can decide if they will accept external PR's to change specific pars over.

**Pull request closed without merging**